src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -38,6 +50,78 @@ void setUserStore(UserStore userStore) {
     this.userStore = userStore;
   }
 
+  /**
+   * Sets the ConversationStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setConversationStore(ConversationStore conversationStore) {
+    this.conversationStore = conversationStore;
+  }
+
+  /**
+   * Sets the MessageStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
+  }
+
+  /**
+   * Loops through the messages and users and counts which user has sent the
+   * most messages in all conversations.
+   * @return String with the username of the most active user
+   */
+  private String getMostActiveUser() {
+    List<Message> messages = messageStore.getAllMessages();
+    List<User> users = userStore.getAllUsers();
+    int currentCount = 0;
+    int maxCount = 0;
+    String mostActiveUser =  null;
+    for (User user : users) {
+      currentCount=0;
+      for(Message message : messages){
+        String author = UserStore.getInstance()
+         .getUser(message.getAuthorId()).getName();
+         if(author.equals(user.getName())){
+           currentCount++;
+           if(currentCount > maxCount){
+             maxCount = currentCount;
+             mostActiveUser = author;
+           }
+         }
+       }
```

Comment:
It would be nice to have this separately as "GetMessagesFromUser" in the "MessageStore", this is helpful because @Dnae-ferguson will also need it, and it will be easier to make changes in future, maybe for future performance optimizations.

Reply:
@GustavoMeza good idea!